tiledb/sm/query/result_tile.cc

```diff
@@ -0,0 +1,221 @@
+/**
+ * @file   result_tile.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2017-2020 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file implements class ResultTile.
+ */
+
+#include "tiledb/sm/query/result_tile.h"
+#include "tiledb/sm/array_schema/dimension.h"
+#include "tiledb/sm/array_schema/domain.h"
+
+#include <cassert>
+#include <iostream>
+#include <list>
+
+namespace tiledb {
+namespace sm {
+
+/* ****************************** */
+/*   CONSTRUCTORS & DESTRUCTORS   */
+/* ****************************** */
+
+ResultTile::ResultTile(
+    unsigned frag_idx, uint64_t tile_idx, const Domain* domain)
+    : domain_(domain)
+    , frag_idx_(frag_idx)
+    , tile_idx_(tile_idx) {
+  assert(domain != nullptr);
+  coord_tiles_.resize(domain->dim_num());
+}
+
+/* ****************************** */
+/*               API              */
+/* ****************************** */
+
+bool ResultTile::operator==(const ResultTile& rt) const {
+  return frag_idx_ == rt.frag_idx_ && tile_idx_ == rt.tile_idx_;
+}
+
+uint64_t ResultTile::cell_num() const {
+  if (!coord_tiles_[0].second.first.empty())
+    return coord_tiles_[0].second.first.cell_num();
+
+  if (!coords_tile_.first.empty())
+    return coords_tile_.first.cell_num();
+
+  if (!attr_tiles_.empty())
+    return attr_tiles_.begin()->second.first.cell_num();
+
+  return 0;
+}
+
+void ResultTile::erase_tile(const std::string& name) {
```

Comment:
we could consider having this return a `Status` so that it could return a non-ok status if `name` isn't found in the coord tiles or attribute tiles.